Load order

deadelf79 opened the pull request:
Добавлена загрузка файлов в определенном порядке (для загрузки зависимостей)
Обновлен файл README - добавил инструкцию для этой загрузки

deadelf79 committed:
Загрузка файлов в определенном порядке

Вначале загружается файл load.cfg, если он есть в директории
Пробегаем по всем строчкам и реквестим файлы из папки, имена которых указаны в строках
Удаляем эти же файлы из списка, чтобы не загружать их второй раз (хотя, по идее, requir'у пофигу на это)
После этого закрываем чтение load.cfg и реквестим все файлы, которые не были указаны в нем, но есть в директории

deadelf79 committed:
Update README.md

Добавил инструкцию по загрузке файлов в определенном порядке

godsent commented on side_script_loader.rb:
```diff
@@ -129,13 +129,30 @@ def pathes
+    dir_entries = Dir.entries(dirname).delete_if{|entry|entry=~/^\./}
+    if dir_entries.include?("load.cfg")
+      load = load_config
+      load.each{|entry|
```
Большое спасибо за request. Прошу поправить теперь поправить немного согласно стайлгайда https://github.com/bbatsov/ruby-style-guide. Для многострочных блоков лучше использовать do ... end форму

godsent commented on side_script_loader.rb:
```diff
@@ -129,13 +129,30 @@ def pathes
+      load.each{|entry|
+        entry.gsub!(/[\n]$/){""}
+        require filename(entry) if entry =~ /\.rb\Z/
+        dir_entries-=["#{entry}.rb"]
```
вставь пробелы перед и после -=

godsent commented on side_script_loader.rb:
```diff
@@ -129,13 +129,30 @@ def pathes
   end
 
+  def load_config
+    open(config,"r").readlines.delete_if{|line|line=~/^[\s]?#/}
```
нужны пробелы:
1. После запятой после первого аргумента
2. Перед блоком 
3. Перед вертикальной чертой в блоке (где принимаются аргументы)
4. После вертикальной черты в блоке
5. Два пробела вокруг =~ 
6. Перед закрывающей скобкой блока

deadelf79 committed:
Исправил оформление кода

deadelf79 committed:
Независимость от расширения

Убрал требование писать расширение подгружаемых скриптов в текстовом файле, эта проверка - лишняя.

deadelf79 commented:
Исправил ошибки оформления, а также убрал один из собственных косяков, который мешал работать (см. коммит 024b7f5)
